dnf5daemon-server/services/base/base.cpp

```diff
@@ -34,14 +34,30 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 void Base::dbus_register() {
     auto dbus_object = session.get_dbus_object();
     dbus_object->registerMethod(
-        dnfdaemon::INTERFACE_BASE, "read_all_repos", "", "b", [this](sdbus::MethodCall call) -> void {
+        dnfdaemon::INTERFACE_BASE, "read_all_repos", "", {}, "b", {"result"}, [this](sdbus::MethodCall call) -> void {
             session.get_threads_manager().handle_method(*this, &Base::read_all_repos, call, session.session_locale);
         });
-    dbus_object->registerSignal(dnfdaemon::INTERFACE_REPO, dnfdaemon::SIGNAL_REPO_KEY_IMPORT_REQUEST, "ossssx");
-    dbus_object->registerSignal(dnfdaemon::INTERFACE_BASE, dnfdaemon::SIGNAL_DOWNLOAD_ADD_NEW, "os");
-    dbus_object->registerSignal(dnfdaemon::INTERFACE_BASE, dnfdaemon::SIGNAL_DOWNLOAD_PROGRESS, "ott");
-    dbus_object->registerSignal(dnfdaemon::INTERFACE_BASE, dnfdaemon::SIGNAL_DOWNLOAD_END, "o");
-    dbus_object->registerSignal(dnfdaemon::INTERFACE_BASE, dnfdaemon::SIGNAL_DOWNLOAD_MIRROR_FAILURE, "o");
+
+    dbus_object->registerSignal(
+        dnfdaemon::INTERFACE_BASE,
+        dnfdaemon::SIGNAL_DOWNLOAD_ADD_NEW,
+        "ossx",
+        {"session_object_path", "download_id", "description", "total_to_download"});
+    dbus_object->registerSignal(
+        dnfdaemon::INTERFACE_BASE,
+        dnfdaemon::SIGNAL_DOWNLOAD_PROGRESS,
+        "osxx",
+        {"session_object_path", "download_id", "total_to_download", "downloaded"});
+    dbus_object->registerSignal(
+        dnfdaemon::INTERFACE_BASE,
+        dnfdaemon::SIGNAL_DOWNLOAD_END,
+        "osis",
```

Comment:
The signature should  be "osus". A code in callbacks.cpp emits uint32_t as well as org.rpm.dnf.v0.Base.xml documents "u" type.